src/CloudFoundryApiClient/test/CfApiClientTestSupport.cs

```diff
@@ -272,4 +293,61 @@ public FakeAppsResponse(string apiAddress, int pageNum, int totalResults, int to
             Apps = apps;
         }
     }
+
+    internal class FakeBuildpacksResponse : BuildpacksResponse
+    {
+        public FakeBuildpacksResponse(string apiAddress, int pageNum, int totalResults, int totalPages, int resultsPerPage)
+        {
+            bool isFirstPage = pageNum == 1;
+            bool isLastPage = pageNum == totalPages;
+
+            var firstHref = new HypertextReference() { Href = $"{apiAddress}{CfApiClient.ListBuildpacksPath}?page=1&per_page={resultsPerPage}" };
+            var lastHref = new HypertextReference() { Href = $"{apiAddress}{CfApiClient.ListBuildpacksPath}?page={totalPages}&per_page={resultsPerPage}" };
+            var nextHref = isLastPage ? null : new HypertextReference() { Href = $"{apiAddress}{CfApiClient.ListBuildpacksPath}?page={pageNum + 1}&per_page={resultsPerPage}" };
+            var previousHref = isFirstPage ? null : new HypertextReference() { Href = $"{apiAddress}{CfApiClient.ListBuildpacksPath}?page={pageNum - 1}&per_page={resultsPerPage}" };
+
+            Pagination = new Pagination
+            {
+                Total_results = totalResults,
+                Total_pages = totalPages,
+                First = firstHref,
+                Last = lastHref,
+                Next = nextHref,
+                Previous = previousHref,
+            };
+
+            Buildpack[] buildpacks;
+
+            var numPreviousResults = (pageNum - 1) * resultsPerPage;
+            
+            if (isLastPage)
+            {
+                int numResourcesInLastPage = totalResults % resultsPerPage;
+                buildpacks = new Buildpack[numResourcesInLastPage];
+
+                for (int i = 0; i < numResourcesInLastPage; i++)
+                {
+                    buildpacks[i] = new Buildpack
+                    {
+                        Name = $"fakeBuildpack{i / 3 + 1 + numPreviousResults}",
```

Comment:
Beware use of "magic" values.  

See:
https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants

Note on precendence ... yes, the division will happen before the addition, however it's useful to include the redundant parens in this particular case because it helps a reader to quickly parse the expression.

Ie, 
```
fakeBuildpack{(i / 3) + 1 + numPreviousResults}
```

Reply:
Addressed in 6107a82